src/wp-includes/class-wp-image-editor.php

```diff
@@ -486,7 +505,7 @@ public function get_suffix() {
 	public function maybe_exif_rotate() {
 		$orientation = null;
 
-		if ( is_callable( 'exif_read_data' ) && 'image/jpeg' === $this->mime_type ) {
+		if ( is_callable( 'exif_read_data' ) ) {
```

Comment:
@adamsilverstein Is it okay to just remove this here? Why was this limited to `image/jpeg` in the first place?

To be clear, I can't see why this would be here in the first place, but since it is, we should probably look back at why this was historically added and whether it's really okay to remove this limitation now.

Reply:
I reviewed briefly. I believe this first appears for tis ticket: https://core.trac.wordpress.org/ticket/14459#comment:93 

I see why errors are silenced - "exif_read_data can throw an error for some jpegs"

Reviewing the code and comments from @pbiron like this one, make me conclude the only reason trying to get EXIF data to JPEGs is that that that was the only format that supported EXIF Orientation data, and that was the narrow focus of the work. Now we have WebP that supports Orientation, and in theory HEIC, AVIF and others.

We could  narrow this change by making it:

```php if ( is_callable( 'exif_read_data' ) && ( 'image/jpeg' === $this->mime_type || 'image/webp' === $this->mime_type )```

I don't see much risk in this change - still, part of the underlying issue here is that we are "setting" the mime type to `image/webp` before secondary mime generation - so even if we know the orientation from the original image upload, the logic skips rotating the secondary type because of the "image" mime not being jpeg. if we limited the changes from `set_mime` to the `output_mime` field we might avoid this issue in the first place, I'll explore that a bit - I'm not sure we tried that.

Reply:
@adamsilverstein Thanks for looking into this. I would err on the site of caution and maybe just change this for now like you've suggested in your answer, checking for either `image/jpeg` or `image/webp`. That would support our use-case better.

But I'm also not opposed to keeping this as you have currently if you're confident about removing the limitation entirely.

> if we limited the changes from `set_mime` to the `output_mime` field we might avoid this issue in the first place, I'll explore that a bit - I'm not sure we tried that.

I think that's overall a great idea potentially too, though I would suggest to keep that logic untouched and rather store the _original_ mime type set in the class in another class property like `$original_mime_type`. This way we could always check for the original mime type, so we could stick to just `image/jpeg` (because of only checking the original mime type) and it would make it also a bit more flexible for other types of image mime transformations.

Reply:
missed this comment before: I'm doing just this - storing the original mime type; tis line now remains unchanged. We can revisit later if this should support more types.